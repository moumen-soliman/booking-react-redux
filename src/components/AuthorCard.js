import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class AuthorCard extends Component {
    render() {
        const {id, name} = this.props
        return (
            <li><Link to={`/author/${id}`}>{name}</Link></li>
        )
    }
}

export default AuthorCard;