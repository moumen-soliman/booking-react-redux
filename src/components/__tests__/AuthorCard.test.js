import React from 'react';
import * as renderer from 'react-test-renderer';
import { MemoryRouter} from 'react-router';
import AuthorCard from '../AuthorCard';

describe('Author Card', () => {
    test('should render link output', () => {
        const component = renderer.create(
            <MemoryRouter> 
                <AuthorCard />
            </MemoryRouter>
        ).toJSON();

        expect(component).toMatchSnapshot();
    });
});
