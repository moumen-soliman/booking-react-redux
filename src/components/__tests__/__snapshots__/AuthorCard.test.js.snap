// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Card should render link output 1`] = `
<li>
  <a
    href="/author/undefined"
    onClick={[Function]}
  />
</li>
`;

exports[`AuthorCard renders correctly 1`] = `<li />`;

exports[`SampleComponent should render 1`] = `
<li>
  <a
    href="/author/undefined"
    onClick={[Function]}
  />
</li>
`;
